RichardSzalay.MockHttp.Shared/MockedRequestExtensions.cs

```diff
@@ -14,16 +15,17 @@ namespace RichardSzalay.MockHttp
     /// Provides extension methods for <see cref="T:MockedRequest"/>
     /// </summary>
     public static class MockedRequestExtensions
-    {
-        /// <summary>
-        /// Constraints the request using custom logic
-        /// </summary>
-        /// <param name="source">The source mocked request</param>
-        /// <param name="matcher">The delegate that will be used to constrain the request</param>
-        /// <returns>The <see cref="T:MockedRequest"/> instance</returns>
-        public static MockedRequest With(this MockedRequest source, Func<HttpRequestMessage, bool> matcher)
-        {
-            return source.With(new CustomMatcher(matcher));
+    {
+		/// <summary>
+		/// Constraints the request using custom logic
+		/// </summary>
+		/// <param name="source">The source mocked request</param>
+		/// <param name="matcher">The delegate that will be used to constrain the request</param>
+		/// <param name="matcherExpression">This will be automatically populated with the text of the matcher when using C# 10</param>
+		/// <returns>The <see cref="T:MockedRequest"/> instance</returns>
+		public static MockedRequest With(this MockedRequest source, Func<HttpRequestMessage, bool> matcher, [CallerArgumentExpression("matcher")] string matcherExpression = "")
```

Comment:
Can we make this an overload? Otherwise it's breaking the public API.

Reply:
Yeah, happy to revert the change to With. 
Given that CallerArgumentExpression needs an empty string default parameter to be able to populate, I think we might need to add a new method (WithExpression?), as without, I don't think the compiler will ever choose the overload new overload.